clmm/utils.py

```diff
@@ -459,3 +459,192 @@ def validate_argument(loc, argname, valid_type, none_ok=False, argmin=None, argm
+    r"""Mean value of the geometric lensing efficicency ratio 
+    
+    .. math:: 
+       \left\<beta_s\right\> =\frac{\sum_{z = z_{min}}^{z = z_{max}}\beta_s(z)p(z)}{\sum_{z = z_{min}}^{z = z_{max}}p(z)}
+
+    Parameters
+    ----------
+    z_cl: float
+            Galaxy cluster redshift
+    z_inf: float
+            Redshift at infinity
+    pdz: 1 argument function
+            Redshift probability density function. Default is\
+            Chang et al (2013) unnormalized galaxy redshift distribution\
+            function.
+    zmin: float
+            Minimum redshift to be set as the source of the galaxy\
+            when performing the sum.
+    zmax: float
+            Minimum redshift to be set as the source of the galaxy\
+            when performing the sum.
+    delta_z_cut: float
+            Redshift interval to be summed with the galaxy redshift to return\
+            a $zmin$ if not provided by the user.
+    cosmo: Cosmology
+        Cosmology object
+    
+    Returns
+    -------
+    float
+        Mean value of the geometric lensing efficicency ratio
+    """   
+    if pdz == None:
+        alpha, beta, redshift0 = 1.24, 1.01, 0.51
+
+        def pdz(z):
+            return (z**alpha)*np.exp(-(z/redshift0)**beta)
+
+    def integrand(z_i, z_cl=z_cl, z_inf=z_inf, cosmo=cosmo):
+        return compute_beta_s(z_i, z_cl, z_inf, cosmo) * pdz(z_i)
+
+    if zmin==None:
+        zmin = z_cl + delta_z_cut
+    Bs_mean = quad(integrand, zmin, zmax)[0] / quad(pdz, zmin, zmax)[0]
+    return Bs_mean
+
+def compute_Bs_square_mean(z_cl, z_inf, cosmo, zmax=4.0, delta_z_cut=0.1, zmin=None, pdz=None):
+    r"""Mean square value of the geometric lensing efficicency ratio 
+    
+    .. math:: 
+       \left\<beta_s\right\>2 =\frac{\sum_{z = z_{min}}^{z = z_{max}}\beta_s^2(z)p(z)}{\sum_{z = z_{min}}^{z = z_{max}}p(z)}
+
+    Parameters
+    ----------
+    z_cl: float
+            Galaxy cluster redshift
+    z_inf: float
+            Redshift at infinity
+    pdz: 1 argument function
+            Redshift probability density function. Default is\
+            Chang et al (2013) unnormalized galaxy redshift distribution\
+            function.
+    zmin: float
+            Minimum redshift to be set as the source of the galaxy\
+            when performing the sum.
+    zmax: float
+            Minimum redshift to be set as the source of the galaxy\
+            when performing the sum.
+    delta_z_cut: float
+            Redshift interval to be summed with the galaxy redshift to return\
+            a $zmin$ if not provided by the user.
+    cosmo: Cosmology
+        Cosmology object
+    
+    Returns
+    -------
+    float
+        Mean square value of the geometric lensing efficicency ratio.
+    """ 
+    if pdz == None:
+        alpha, beta, redshift0 = 1.24, 1.01, 0.51
+        def pdz(z):
+            return (z**alpha)*np.exp(-(z/redshift0)**beta)
```

Comment:
Do we want to move (or import?) _chang_z_distrib from support/mock_data.py to utils.py so we don't have to define it again here?

Reply:
I agree we want them in one place only. But actually, I'd go the other way around. Have the pdf defined in `utils` and import them to `mock_data`.

Reply:
Agree. Do we still keep it as a private function? or do we want to make it a public function?

Reply:
I don't have a strong opinion, but can't think of any objection to making it public.